tianshou/policy/modelfree/bdq.py

```diff
@@ -0,0 +1,164 @@
+from typing import Any, Dict, Optional, Union
+
+import numpy as np
+import torch
+
+from tianshou.data import Batch, ReplayBuffer, to_numpy, to_torch_as
+from tianshou.policy import DQNPolicy
+from tianshou.utils.net.common import BranchingNet
+
+
+class BranchingDQNPolicy(DQNPolicy):
+    """Implementation of the Branching dual Q network arXiv:1711.08946.
+
+    :param torch.nn.Module model: a model following the rules in
+        :class:`~tianshou.policy.BasePolicy`. (s -> logits)
+    :param torch.optim.Optimizer optim: a torch.optim for optimizing the model.
+    :param float discount_factor: in [0, 1].
+    :param int estimation_step: the number of steps to look ahead. Default to 1.
+    :param int target_update_freq: the target network update frequency (0 if
+        you do not use the target network). Default to 0.
+    :param bool reward_normalization: normalize the reward to Normal(0, 1).
+        Default to False.
+    :param bool is_double: use double network. Default to True.
+
+    .. seealso::
+
+        Please refer to :class:`~tianshou.policy.BasePolicy` for more detailed
+        explanation.
+    """
+
+    def __init__(
+        self,
+        model: BranchingNet,
+        optim: torch.optim.Optimizer,
+        discount_factor: float = 0.99,
+        estimation_step: int = 1,
+        target_update_freq: int = 0,
+        reward_normalization: bool = False,
+        is_double: bool = True,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(
+            model, optim, discount_factor, estimation_step, target_update_freq,
+            reward_normalization, is_double
+        )
+        assert estimation_step == 1, "N-step bigger than one is not supported by BDQ"
+        self.max_action_num = model.action_per_branch
+        self.num_branches = model.num_branches
+
+    def _target_q(self, buffer: ReplayBuffer, indices: np.ndarray) -> torch.Tensor:
+        batch = buffer[indices]  # batch.obs_next: s_{t+n}
+        result = self(batch, input="obs_next")
+        if self._target:
+            # target_Q = Q_old(s_, argmax(Q_new(s_, *)))
+            target_q = self(batch, model="model_old", input="obs_next").logits
+        else:
+            target_q = result.logits
+        if self._is_double:
+            act = np.expand_dims(self(batch, input="obs_next").act, -1)
+            act = torch.from_numpy(act)
+            if not target_q.get_device() < 0:  # TODO better?
+                act = act.to(target_q.get_device())
```

Comment:
Why not use
```suggestion
            act = to_torch_as(act, target_q)
```
Same as other place.